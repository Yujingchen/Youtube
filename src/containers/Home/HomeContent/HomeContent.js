import React, { Component } from "react";
import VideoGrid from "../../../components/VideoGrid/VideoGrid";
import "./HomeContent.scss";
export default class HomeContent extends Component {
  render() {
    return (
      <div className="home">
        <div className="responsive-video-grid-container">
          <VideoGrid title="trending" />
          <VideoGrid title="Autos & Vehicles" hideDivider={true} />
        </div>
      </div>
    );
  }
}
