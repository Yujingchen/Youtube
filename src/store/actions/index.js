export function createAction(type, payload = {}) {
  return {
    type,
    ...payload
  };
}

export const REQUEST = "REQUEST";
export const SUCCESS = "SUCCESS";
export const FAILURE = "FAILURE";

export function createRequestTypes(base) {
  if (!base) {
    throw new Error("cannot create request type with base = '' or base = null");
  }
  return [REQUEST, SUCCESS, FAILURE].reduce((acc, type) => {
    acc[type] = `${base}_${type}`;
    return acc;
  }, {});
}

//RETURN A OBJECT WITH THREE FIELDS
// {
//   REQUEST: 'MOST_POPULAR_REQUEST',
//   SUCCESS: 'MOST_POPULAR_SUCCESS',
//   FAILURE: 'MOST_POPULAR_FAILURE'
// }
