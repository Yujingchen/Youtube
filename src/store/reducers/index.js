import apiReducer from "./api";
import { combineReducers } from "redux";
import videosReducer from "./video";
export default combineReducers({
  api: apiReducer,
  videos: videosReducer
});

export function ignoreErros(fn, ...args) {
  return () => {
    const ignoreErrorCallback = response => response;
    return fn(...args).then(ignoreErrorCallback, ignoreErrorCallback);
  };
}
