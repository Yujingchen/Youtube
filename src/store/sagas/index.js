import { all, call, put, fork } from "redux-saga/effects";
import {
  watchMostPopularVideos,
  watchMostPopularVideosByCategory,
  watchVideoCategories
} from "./video";
import { watchWatchDetails } from "./watch";
import { watchSearchForVideos } from "./search";

export default function*() {
  yield all([
    fork(watchMostPopularVideos),
    fork(watchVideoCategories),
    fork(watchMostPopularVideosByCategory),
    fork(watchWatchDetails),
    fork(watchSearchForVideos)
  ]);
}
//run watcher saga for listening to certian type of action
//yield
/*
... rest unchange for now
*/

/*
 * entity must have a success, request and failure method
 * request is a function that returns a promise when called
 * */
export function* fetchEntity(request, entity, ...args) {
  try {
    const response = yield call(request);
    //make api call and save the response
    // we directly return the result object and throw away the headers and the status text here
    // if status and headers are needed, then instead of returning response.result, we have to return just response.
    yield put(entity.success(response.result, ...args));
  } catch (error) {
    yield put(entity.failure(error, ...args));
  }
}

export function ignoreErrors(fn, ...args) {
  return () => {
    const ignoreErrorCallback = response => response;
    return fn(...args).then(ignoreErrorCallback, ignoreErrorCallback);
  };
}
