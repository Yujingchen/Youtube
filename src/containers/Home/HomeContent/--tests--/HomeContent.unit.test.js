import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { configureStore } from "../../../../store/configureStore";
import HomeContent from "../HomeContent";
const store = configureStore();

describe("HomeContent", () => {
  test("renders", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <HomeContent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
