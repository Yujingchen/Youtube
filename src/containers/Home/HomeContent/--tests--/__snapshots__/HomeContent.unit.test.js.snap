// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent renders 1`] = `
<div
  className="home"
>
  <div
    className="responsive-video-grid-container"
  >
    <VideoGrid
      title="trending"
    />
    <VideoGrid
      hideDivider={true}
      title="Autos & Vehicles"
    />
  </div>
</div>
`;
